lib/Option/SanitizerOptions.cpp

```diff
@@ -258,6 +258,22 @@ OptionSet<SanitizerKind> swift::parseSanitizerRecoverArgValues(
   return sanitizerRecoverSet;
 }
 
+bool swift::parseSanitizerAddressUseODRIndicator(
+    const llvm::opt::Arg *A, const OptionSet<SanitizerKind> &enabledSanitizers,
```

Comment:
Given that we don't expect this to be `NULL` (see below), I think that it would be better to change the signature to take the `Arg` by reference (`const llvm::opt::Arg &A`).

Reply:
This is a valid criticism but the use of `llvm::opt::Arg*` here is deliberate. It's being used so that this new function follows a similar design as the other functions in `include/swift/Option/SanitizerOptions.h`.

I'd prefer to keep the APIs as uniform as possible. Note I'm aware there are some oddities already in this file (inconsistent argument order) but we shouldn't add to them if we don't have to again.

Reply:
I don't think its adding any inconsistency.  The LLVM style actually encourages this - pass by reference unless you need to be able to pass in `nullptr` as a reference to `nullptr` is not well-formed.

Reply:
I wasn't talking about inconsistency with respect to LLVM's recommended style. I'm talking about inconsistency with respect to the APIs that already exist in `SanitizerOptions.h`. They **all** take a `llvm::opt::Arg*`. All of these could actually be `Arg&` because they all assume that `Arg* A` is not null. Deliberately adding an API that doesn't match the existing APIs is an unnecessary inconsistency. We could change all of the functions in the header but that's really out-of-scope for this PR. If you really want this that let's do that in a separate PR.

I suspect the reason these APIs take a `llvm::opt::Arg*` right now is that is the type available at the use sites and so passing it directly was the most convenient thing to do.